src/main.rs

```diff
@@ -47,125 +97,293 @@ impl LogLevel {
 }
 
 #[derive(Subcommand)]
+/// params that greenboot accepts
+///
+/// greenboot health-check -> runs the custom health checks
+///
+/// greenboot rollback -> if boot_counter satisfies it trigger rollback
 enum Commands {
-    Check,
-    Stamp,
-}
-
-#[derive(Serialize, Deserialize, PartialEq, Eq, Hash)]
-struct ServiceStatus {
-    unit: String,
+    HealthCheck,
+    Rollback,
 }
 
-fn check() -> Result<(), Error> {
-    match File::open("/etc/greenboot/upgrade.stamp") {
-        Ok(_) => {
-            log::info!("stamp on disk, removing and running greenboot");
-            std::fs::remove_file("/etc/greenboot/upgrade.stamp")?
+/// runs all the scripts in required.d and wanted.d
+fn run_diagnostics() -> Result<()> {
+    let mut required_script_failure: bool = false;
+    let mut path_exists: bool = false;
+    for path in GREENBOOT_INSTALL_PATHS {
+        let greenboot_required_path = format!("{path}/check/required.d/");
+        if !Path::new(&greenboot_required_path).is_dir() {
+            continue;
         }
-        Err(e) => match e.kind() {
-            ErrorKind::NotFound => return Ok(()),
-            _ => {
-                bail!("unknown error when opening stamp file: {:?}", e);
-            }
-        },
-    }
-    let mut failure = false;
-    for path in [
-        "/usr/lib/greenboot/check/required.d/*.sh",
-        "/etc/greenboot/check/required.d/*.sh",
-    ] {
-        for entry in glob(path)?.flatten() {
-            log::info!("running required check {}", entry.to_string_lossy());
-            let output = Command::new("bash").arg("-C").arg(entry).output()?;
-            if !output.status.success() {
-                // combine and print stderr/stdout
-                log::warn!("required script failed...");
-                failure = true;
+        path_exists = true;
+        if let Some(errors) = run_scripts("required", &greenboot_required_path).err() {
```

Comment:
see comment about the `run_scripts` function, but in here we could just simplify things and just check if the vector is empty or not